src/microservices/Microservices.IsIdentifiable/Rules/IsIdentifiableRule.cs

```diff
@@ -43,17 +42,13 @@ public class IsIdentifiableRule : ICustomRule
         public string IfPattern
         {
             get => _ifPattern?.ToString();
-            set => _ifPattern = value == null ? null : new Regex(value, RegexOptions.IgnoreCase | RegexOptions.Compiled);
+            set => _ifPattern = value == null ? null : new Regex(value, (IfPatternIsCaseSensitive ? 0 : RegexOptions.IgnoreCase) | RegexOptions.Compiled);
         }
+
         /// <summary>
-        /// The Regex pattern which should be used to match values with
+        /// Whether the IfPattern match is case sensitive or not (default is false)
         /// </summary>
-        public string IfPatternCaseSensitive
-        {
-            get => _ifPattern?.ToString();
-            set => _ifPattern = value == null ? null : new Regex(value, RegexOptions.Compiled);
-        }
-
+        public bool IfPatternIsCaseSensitive  { get; set; }
```

Comment:
Just `CaseSensitive` would probably be enough for a name?  We should be able to use a backing private variable _caseSensitive then on `set` we can change `_ifPattern` to a new Regex? that way we can support changing the sensitivity on the object and not have to worry about order of property setting in yaml deserialization (which may not nessesarily match how it is read from the file necessarily?  Also would be unexpected with the current codebase that the set does nothing unless youa lso change the IfPattern immediately afterwards?

Reply:
Ok that makes sense.  I can make the change if you want and add some unit tests, its a nice self contained class so should be easy to test? Definetly think its worth the effort, if a class has a public set it should work toggling it at any state of the object (or throw appropriate exceptions)